realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/CredentialsTests.kt

```diff
@@ -51,15 +58,21 @@ class CredentialsTests {
     @Test
     fun anonymous() {
         val creds = Credentials.anonymous()
-        assertEquals("anon-user", creds.identityProvider.id)
+        assertEquals(Credentials.IdentityProvider.ANONYMOUS.id, creds.identityProvider.id)
         assertTrue(creds.asJson().contains("anon-user")) // Treat the JSON as an opaque value.
     }
 
-    @Ignore("FIXME: Awaiting ObjectStore support")
     @Test
     fun apiKey() {
         val creds = Credentials.apiKey("token")
-        assertEquals("anon-user", creds.identityProvider.id)
+        assertEquals(Credentials.IdentityProvider.SERVER_API_KEY.id, creds.identityProvider.id)
```

Comment:
This looks wrong... `SERVER_API_KEY` should be the test below. This should be `API_KEY`. A bit concerning if the test actually pass? 🤔

Reply:
Hmmm, this smells like a copy-paste error to me, but you're right, looks odd. I'll investigate further.

Reply:
Ahhh, I know why. It's because `API_KEY` and `SERVER_API_KEY` have the same id value as per the OS implementation:

```
IdentityProvider const IdentityProviderUserAPIKey = "api-key";
IdentityProvider const IdentityProviderServerAPIKey = "api-key";
```

🤷

I remember I mentioned it and we agreed it was a bit silly, but we needed to keep it separate because of the OS implementation, although being technically the same thing, treats them as two different identity providers. That's why the test passes.